### PR TITLE
error in dev mode if global fetch is used with relative URL

### LINKED ISSUE #7976: Attempt to error better when wrong fetch is used on an endpoint
### Describe the problem

I've now twice gotten burned by using the "wrong" fetch in my loaders. Running

fetch("/api/my-endpoint")

with the global fetch that's in scope will puke, since "/api/my-endpoint" has no way of resolving. Obviously you have to destructure the fetch that's passed to your loader. Simple enough, but it's an easy thing to miss when you're new (at SvelteKit), and it's *maddening* to debug.

### Describe the proposed solution

Would it be possible for the global fetch to detect when an (otherwise valid) api endpoint is being requested, and error with something clear about needing to use the provided fetch to hit endpoints?

I know this woulnd't be foolproof, since shared loaders are (I assume?) using the browser's provided fetch, which would be hard for you all to wrap, but on the server you're already providing your own fetch, so could this check be added to that (presumably for non-production builds)? 

### Alternatives considered

n/a

### Importance

nice to have

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Rich-Harris**
closes #7976

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Review by Conduitry — CHANGES_REQUESTED** [review]
What happens if someone uses a protocol-relative URL like `//foo/bar`? This check would fail to catch it and it would (I think) then still fail with the global `fetch`.

Can we instead just check something simpler like that the URL matches `/^\w+:\/\//`? Or heck, just hard code `http://` and `https://` since I don't think anything else would succeed anyway?

File: packages/kit/src/exports/vite/dev/index.js [hunk 1]
```diff
@@ -29,6 +29,19 @@ export async function dev(vite, vite_config, svelte_config) {
...
 
+	const fetch = globalThis.fetch;
+	globalThis.fetch = (info, init) => {
+		if (typeof info === 'string') {
+			if (new URL(info, 'fake://fake').protocol === 'fake:') {
+				throw new Error(
+					`Cannot use relative URL (${info}) with global fetch — use \`event.fetch\` instead: https://kit.svelte.dev/docs/web-standards#fetch-apis`
+				);
+			}
+		}
```
Comment (Rich-Harris) [inline packages/kit/src/exports/vite/dev/index.js:40]: probably simpler

```suggestion
		if (typeof info === 'string' && !/^\w+:\/\//.test(info)) {
			throw new Error(
				`Cannot use relative URL (${info}) with global fetch — use \`event.fetch\` instead: https://kit.svelte.dev/docs/web-standards#fetch-apis`
			);
		}
```

**Comment by Rich-Harris** [comment]
The check would work — `new URL('//foo.com', 'fake://fake').protocol === 'fake:'`. But yeah, a regex is probably simpler. I opened a suggestion

**Comment by Conduitry** [comment]
Oh, whoops, not sure what I was thinking there anymore. I do still like the regex though.

**Review by dummdidumm — APPROVED** [review]
Approving the regex approach. Hardcoding http(s) would make things like tauri apps etc impossible to use the global fetch

**Pull request merged**